server/src/routes/cards.js

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable no-unused-vars */
+var express = require("express");
+const res = require("express/lib/response");
+var router = express.Router();
+const cardController = require("../controller/cardController");
+
+router.get("/", async function (_req, res, _next) {
+	res.send(await cardController.listCards());
+});
+router.get("/:cardType", async function (req, res, _next) {
+	res.send(await cardController.listCards(req.params.cardType));
+});
+router.post("/new", async function (req, res, _next) {
+	res.send(await cardController.insertCard(req.body));
+});
+router.post("/update", async function (req, _res, _next) {
```

Comment:
Existe uma erro semântico aqui, o método post é usado para criar recurso e o path me faz entender que você quer altera-lo. Acredito que essa rota poderia ser definida dessa forma:

```suggestion
router.put("/:cardId", async function (req, _res, _next) {
```

Referência
https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Methods